### PR TITLE
 Replace xxHash with blake2b

### PR DESCRIPTION
**Pull request by SergiySW**

### CODE REVIEW

File: nano/node/common.hpp [hunk 1]
```diff
@@ -22,40 +21,57 @@ bool parse_tcp_endpoint (std::string const &, nano::tcp_endpoint &);
 
 namespace
 {
+uint64_t random_64_m{ 0 };
+uint64_t const & random_64 ()
+{
+	if (random_64_m == 0)
+	{
+		random_64_m = (uint64_t) nano::random_pool::generate_word32 (0, 0xffffffffUL) << 32 | nano::random_pool::generate_word32 (0, 0xffffffffUL);
```
Comment (cryptocode) [inline nano/node/common.hpp]: Could we make a `nano::random_pool::generate_word64` ? That would lock the pool mutex only once and everything is in one place.

File: nano/node/common.hpp [hunk 2]
```diff
@@ -22,40 +21,57 @@ bool parse_tcp_endpoint (std::string const &, nano::tcp_endpoint &);
 
 namespace
 {
+uint64_t random_64_m{ 0 };
+uint64_t const & random_64 ()
+{
+	if (random_64_m == 0)
```
[the remark this answers is not in the document]
Comment (clemahieu) [reply nano/node/common.hpp]: Let's put this in to node/ledger constants and make it 128bits.

File: nano/node/common.hpp [hunk 3]
```diff
@@ -22,40 +21,57 @@ bool parse_tcp_endpoint (std::string const &, nano::tcp_endpoint &);
...
 	address.bytes = endpoint_a.address ().to_v6 ().to_bytes ();
-	XXH64_state_t * const state = XXH64_createState ();
-	XXH64_reset (state, 0);
-	XXH64_update (state, address.bytes.data (), address.bytes.size ());
 	auto port (endpoint_a.port ());
-	XXH64_update (state, &port, sizeof (port));
-	auto result (XXH64_digest (state));
-	XXH64_freeState (state);
+	blake2b_state state;
+	blake2b_init (&state, sizeof (result));
```
Comment (clemahieu) [inline nano/node/common.hpp]: Let's use the blake2b_init_key initializer which does the key-initialization for us.

Comment (SergiySW) [reply nano/node/common.hpp]: It's much slower than blake2b_update, like 1.5-2 times

File: nano/node/common.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (SergiySW) [reply nano/node/common.hpp]: @wezrule could mutex kill performance even more?

File: nano/node/common.hpp (hunk 1 above)
Comment (SergiySW) [reply nano/node/common.hpp]: Colin suggested 128 bit, then new function seems not required

**Comment by SergiySW** [comment]
Can be modified with random_constants from https://github.com/nanocurrency/nano-node/pull/1928

**Comment by wezrule** [comment]
I did an ASAN run on core_test/live and it didn't hit any of CryptoPP uninitialized global warnings, so LGTM.

**Pull request merged**